File: 8kyu/(8 kyu) Is the string uppercase/(8 kyu) Is the string uppercase.dart
// 1 Plain solution
bool isUpperCase(String str) {
  for (int i = 0; i < str.length; i++) {
    if (str[i] != str[i].toUpperCase()) {
      return false;
    }
  }

  return true;
}

// 2 Optimized solution
bool isUpperCase(String str) {
  return str.runes.every((int rune) {
    var character = String.fromCharCode(rune);
    return character == character.toUpperCase();
  });
}

// 3 Clever solution
bool isUpperCase(String str) => str == str.toUpperCase();

File: 8kyu/(8 kyu) Is the string uppercase/(8 kyu) Is the string uppercase.ts
// 1 Simple solution
export function isUpperCase(str: string) : boolean {
  for (let i = 0; i < str.length; i++) {
    if (str[i] !== str[i].toUpperCase()) {
      return false;
    }
  }

  return true;
}

// 2 Optimized solutoion
export function isUpperCase(str: string) : boolean {
  return [...str].filter(s => s === s.toUpperCase()).length === str.length;
}

// 3 Clever solution
export function isUpperCase(str: string) : boolean {
  return str === str.toUpperCase();
}

// 4 Coding golf
export const isUpperCase = (s: string) => s === s.toUpperCase()
